xbmc/utils/SysfsUtils.cpp

```diff
@@ -16,105 +16,140 @@
 #include <fcntl.h>
 #include <string.h>
 
-#ifdef TARGET_WINDOWS_STORE
-#include <io.h>
-#endif
+static const unsigned int PAGESIZE = getpagesize();
 
-int SysfsUtils::SetString(const std::string& path, const std::string& valstr)
+bool SysfsUtils::SetString(const std::string& path, const std::string& value)
 {
-  int fd = open(path.c_str(), O_RDWR, 0644);
-  int ret = 0;
-  if (fd >= 0)
+  auto oldValue = GetString(path);
+  if (oldValue == value)
   {
-    if (write(fd, valstr.c_str(), valstr.size()) < 0)
-      ret = -1;
-    close(fd);
+    return true;
```

Comment:
Are you sure this is a good idea? Does the kernel interface guarantee that writing a value that already "seems" to be there never has any (desired) side-effect?